broadlink/switch.py

```diff
@@ -260,19 +260,17 @@ def set_state(
         response = self.send_packet(0x6A, packet)
         return self._decode(response)
 
-    def check_power(self):
+    def check_power(self) -> bool:
         """Return the power state of the device."""
         state = self.get_state()
-        if state:
-            return state["pwr"]
+        return state["pwr"] if state else None
```

Comment:
Why not just `return state["pwr"]`?

Reply:
The original code checked if state is truthy. I imagine that was a safeguard against `get_state()` not returning the expected dict, in which case there wouldn't be any `pwr`to get. `get_state()`, in turn, ultimately depends on `json.loads()`, which on closer inspection shouldn't be able to fail in a way that produce a falsy result, but rather a `JSONDecodeError`. So yes, those lines should be perfectly ok without the safeguard. I'll make them so.

Reply:
I think these functions should return None. If something goes wrong the device will return an error in the response and we raise [here](https://github.com/mjg59/python-broadlink/blob/d8b853d96c9bc64a253e41657789359f6bb165bd/broadlink/switch.py#L305). So there is no reason to return a state we already know.